### PR TITLE
Aya's GSoC final report

### PR DESCRIPTION
**Pull request by aya9aladdin**
This is my GSoC final blog post

### CODE REVIEW

**Comment by hmacdope** [comment]
@MDAnalysis/gsoc-mentors relevant mentors can you review here?

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (hmacdope) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:10]: Can you clairfy that you mean an `MDAnalysis` Universe rather than just a general universe (eg the one in which we live).

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (hmacdope) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:17]: Change
aware -> careful.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (hmacdope) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:19]: Why was it important?

**Review by hmacdope — CHANGES_REQUESTED** [review]
Looking great @aya9aladdin, a nice and complete summary of your project! 
Just a few minor things on my end. I will ask one of the mentors who supervised you to have a look through before merge. :)

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (aya9aladdin) [reply _posts/2022-11-12-Aya-gsoc-final-report.md:19]: I'll clarify it more

**Comment by aya9aladdin** [comment]
I made some updates based on Hugo's sugesstions

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:61]:
```suggestion
### 3. DefaultGuesser class
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:64]:
```suggestion
### 4. Testing the old parser behavior is preserved
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:68]: dividers probably not needed when you use headings

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:13]: I'd remove dividers and rely on headings for structuring.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:21]: This line should not be a level-1 heading as it messes up the document structure (and the indexing for search!).

At the moment I do not see how this line fits into the overall structure. If you follow my advice to make the list items level-3 headings then you can either make it a level-4 heading 
```suggestion
#### Removing types and mass guessing from Topology Parsers ([commit 0cbc497])
```
or just bold face it to draw attention
```suggestion
**Removing types and mass guessing from Topology Parsers ([commit 0cbc497])**
```

I *think* it fits better as a level-4 heading but you need to judge what structure you want.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:27]: I'd make the contexts strings set in normal type (with quotation marks). When typeset as monospace I think of them as objects.

Comment (orbeckst) [reply _posts/2022-11-12-Aya-gsoc-final-report.md:27]: `Guesser` — is it a class name?

Comment (orbeckst) [reply _posts/2022-11-12-Aya-gsoc-final-report.md:27]: The `guess_topologyAttributes` make the following process: -> The `guess_topologyAttributes()` method proceeds through the following steps:

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:29]: I'd make this a numbered list to emphasize sequential steps.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:35]: `guess_attrs()` method

---

(monospaced and I add () after methods/functions)

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:49]: PEP8
```suggestion
     u = mda.Universe(two_water_gro, context='default', to_guess=['bonds'])
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:65]: types and masses are strings: have a default value of `["types", "masses"]`

Comment (orbeckst) [reply _posts/2022-11-12-Aya-gsoc-final-report.md:65]: Make the enumeration explicit (one line for each item) instead of inline: This makes it easier to skim-read the post.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:67]: I also discovered a bug in the `guessed_attributes()` and `read_attributes()` methods of the `Topology` class ...

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:74]: Again, I would resort to level-3 headings
```suggestion
### 1. Elements guessing
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:89]:
```suggestion
### 2. Masses
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:93]: Not needed, I think.
```suggestion
```

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:98]:
```suggestion
```

**Review by orbeckst — COMMENTED** [review]
Looking really good, @aya9aladdin . Good introduction, good detail.

I have a bunch of suggestions, primarily on structuring for presentation as blog post. 

I hope @jbarnoud can give it a quick read, too, so that this can go live soon.

**Comment by aya9aladdin** [comment]
There was some mess in the hierarchy of the text thank @orbeckst  for pointing me to modify them

**Review by hmacdope — APPROVED** [review]
Thanks @aya9aladdin, looks great. I will approve and wait for feedback from others. :)

**Comment by orbeckst** [comment]
ping @jbarnoud — a quick "aye" would be sufficient

(before merging we'd change the date...)

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (jbarnoud) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:10]: A comma is missing at the beginning of the line.

```suggestion
In [MDAnalysis], molecular topologies come from various file formats and forcefields, each of which has its own rules regarding formatting and molecular properties representation. Not all molecular properties are read from files, either because they are simply not provided in the file or because they can be inferred from the nature of the molecular system or forcefield definitions (e.g., for atomic systems masses can be inferred from atom type, while in the Martini forcefield there are only three definitions of masses for its known beads). MDAnalysis had different guessing methods that can infer missing molecular properties for the [Universe]'s topology (eg. [guessing Atom type, mass, and bond]). But those methods suffered from the main issue of being so generic, which causes the guessing output to be inaccurate and not reliable for some topologies as not all topologies speak the same language. That’s why we need to develop tailored context-aware guesser classes to serve different molecular dynamics worlds, in addition, we need to establish a way to pass this context to the [Universe], so the [Universe] uses the appropriate guesser class in guessing topology attributes. 
```

Comment (jbarnoud) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:10]: The issue is not so much that the guessing is too generic. On the contrary, it comes with too many unwritten assumptions.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (jbarnoud) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:41]: An example of how to guess nothing would be nice. Personally, it is my main use for the feature.

**Review by jbarnoud — APPROVED** [review]
Very good summary! I made a couple of comments but it could go forward as is.

**Comment by aya9aladdin** [comment]
> Very good summary! I made a couple of comments but it could go forward as is.

I'll add your updates at the end of this day 👍🎉

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:3]: Change to 
```suggestion
title: "GSoC Report: Adding a new guessing methodology for molecular systems properties in MDAnalysis"
```
The page will not build if there's a `:` outside quotes.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:23]: change to
```suggestion
![Guessing inside Parsers]({{ site.baseurl }}{{ site.images }}/final_report_aya/parsers_guessing.png)
```
Otherwise the PNG will not be found.

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:31]: I would just use the standard markdown formatting for lists
```markdown
1. gets the appropriate Guesser class that matches the passed context.

2. checks if the Guesser class support guessing the attributes passed to the`to_guess` and/or `force_guess` parameters.
```
and then it will look like this:

## Markdown list
image



## Current manually created list

Currently it looks like
image

File: _posts/2022-11-12-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-11-12-Aya-gsoc-final-report.md:77]: I'd also switch to mdarkdown list

**Review by orbeckst — CHANGES_REQUESTED** [review]
@aya9aladdin I built the site locally to check and there are a few things that you _must_ change so that the site builds correctly. (You should be able to simply accept my suggestions.)

Please also change the filename to the day when you make the changes so that it shows up at the top of the blog.

Finally, I have a number formatting suggestions but it's ultimately up to you if you want to change formatting.

You can also explicitly mention your own blog and link to your GSoC blog, e.g., for any readers who want to learn more about your process (e.g., in the "Lessons Learned" section.)

Overall, really nice blog post, well done!

**Comment by aya9aladdin** [comment]
> @aya9aladdin I built the site locally to check and there are a few things that you _must_ change so that the site builds correctly. (You should be able to simply accept my suggestions.)
> 
> Please also change the filename to the day when you make the changes so that it shows up at the top of the blog.
> 
> Finally, I have a number formatting suggestions but it's ultimately up to you if you want to change formatting.
> 
> You can also explicitly mention your own blog and link to your GSoC blog, e.g., for any readers who want to learn more about your process (e.g., in the "Lessons Learned" section.)
> 
> Overall, really nice blog post, well done!

I added all of your suggestions, thanks, Oliver!

File: _posts/2022-12-9-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-12-9-Aya-gsoc-final-report.md:122]: valuable -> valuable

lesson -> lessons

File: _posts/2022-12-9-Aya-gsoc-final-report.md
Comment (orbeckst) [inline _posts/2022-12-9-Aya-gsoc-final-report.md:123]: specially gratful for -> especially grateful to

**Review by orbeckst — APPROVED** [review]

**Comment by orbeckst** [comment]
@aya9aladdin let us know when you think your post is ready to be merged. I think all your reviewers are happy. Any of us can merge it when you give the go signal. 🚀

**Comment by aya9aladdin** [comment]
You can go for sure now 🚀👩‍🚀

**Comment by jbarnoud** [comment]
Merged! Thank you :smile:

**Pull request merged**
